eventhubskafka-flink-eventhubskafka/kubernetes/provision-aks-flink-cluster.sh

```diff
@@ -17,7 +17,13 @@ echo 'creating AKS cluster'
 echo ". name: $AKS_CLUSTER"
 
 if ! az aks show --name $AKS_CLUSTER --resource-group $RESOURCE_GROUP >/dev/null 2>&1; then
-az aks create --name $AKS_CLUSTER --resource-group $RESOURCE_GROUP --node-count $AKS_NODES -s $AKS_VM_SIZE -k $AKS_KUBERNETES_VERSION --generate-ssh-keys -o tsv >> log.txt
+  # if SP vars are set and not empty
+  if [[ ! -z ${AD_SP_APP_ID+x} && ! -z ${AD_SP_SECRET+x} && -n "$AD_SP_APP_ID" && -n "$AD_SP_SECRET" ]]; then
+    echo ". service-principal: $AD_SP_APP_ID"
+    az aks create --name $AKS_CLUSTER --resource-group $RESOURCE_GROUP --node-count $AKS_NODES -s $AKS_VM_SIZE -k $AKS_KUBERNETES_VERSION --generate-ssh-keys --service-principal $AD_SP_APP_ID  --client-secret $AD_SP_SECRET -o tsv >> log.txt
+  else
+    az aks create --name $AKS_CLUSTER --resource-group $RESOURCE_GROUP --node-count $AKS_NODES -s $AKS_VM_SIZE -k $AKS_KUBERNETES_VERSION --generate-ssh-keys -o tsv >> log.txt
```

Comment:
How about incorporating the change from Julien?

```
az ad sp create-for-rbac --skip-assignation
az aks create --service-principal SPN_ID --client-secret SPN_SECRET
```

Reply:
This PR change makes only sense for scenarios where the call `az ad sp create-for-rbac --skip-assignation` has been done upfront (e.g. by a different person/department) as the person that runs the sample might not have permission to create a SP.

The output of the SP create command is what will go into `AD_SP_APP_ID` and `AD_SP_SECRET`.